velox/connectors/hive/storage_adapters/s3fs/S3FileSystem.h

```diff
@@ -26,6 +29,14 @@ bool initializeS3(
 
 void finalizeS3();
 
+using AWSCredentialsProviderFactory =
+    std::function<std::shared_ptr<Aws::Auth::AWSCredentialsProvider>(
+        const S3Config& config)>;
+
+void registerAWSCredentialsProvider(
```

Comment:
Let's add this to the public API registerS3FileSystem.h.

Reply:
@majetideepak I tried moving the declaration into registerS3FileSystem.h, then found it requires to include `<aws/core/auth/AWSCredentialsProvider.h>` and `"velox/connectors/hive/storage_adapters/s3fs/S3Config.h"`, which should be controlled by `VELOX_ENABLE_S3` flags. However, we shouldn't use VELOX_ENABLE_S3 in the header.  Do you have any suggestions?

Reply:
@marin-ma Let's forward declare or create a wrapper for the AwsCredentialsProvider in S3FileSystem.